src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -203,12 +203,12 @@ public override string Description
         {
             get
             {
-                if (DisplayMetadata.Description == null)
+                if (DisplayMetadata.Description != null)
                 {
-                    return null;
+                    return DisplayMetadata.Description();
                 }
 
-                return DisplayMetadata.Description();
+                return null;
```

Comment:
Sorry, I didn't even consider that. They changed because I was reversing the mistake I made about caching the values.